api/document/serializers/doc_cursor.py

```diff
@@ -22,12 +23,27 @@ def to_representation(self, instance: DocCursor) -> List[PrimitiveDict]:
             context={**self.context, 'is_root': False},
         ).data
 
+    def validate_type_emblem_uniqueness(self, data: List[PrimitiveDict]):
+        seen: Set[Tuple[str, str]] = set()
+        for child in data:
+            if 'type_emblem' not in child:
+                continue
```

Comment:
We'd better check that the auto-incrementing logic will prevent a conflict, too. Otherwise, you could imagine input like
```
<parent>
    <child />
    <child emblem="1" />
</parent>
```
That makes me wonder if this check should be happening somewhere else? Maybe in the deserializer?

Reply:
<img src="https://media1.giphy.com/media/xT0xeJpnrWC4XWblEk/giphy.gif">

whoa.

just so i fully understand the problem, should the auto-incrementer work _around_ the potential conflict, or should an error be raised to the user?  I'd assume the former, since otherwise users (or at least editors of AKN XML) would need to understand the auto-incrementing algorithm, but this AKN spec has been full of other weird surprises, so who knows 😆

Reply:
Oh, good point. We should probably trust the user's input here -- if they want to label the first child "2", let's let them.

Reply:
Hehe ok. Do you think I should add that kind of work-around logic in this PR, or a separate one?

Reply:
Separate PR or even kicking the can down the road in a separate issue sounds fine.